src/main/java/org/mitre/hapifhir/BackendAuthorizationInterceptor.java

```diff
@@ -116,15 +130,40 @@ private RSAPublicKey getKey()
   /**
    * Verify the access token signature and expiration.
    * @param token - the access token
-   * @return true if signature is valid and token has not expired, otherwise throws an exception
+   * @return void if signature is valid and token has not expired, otherwise throws an exception
    */
-  private boolean verify(String token) throws IllegalArgumentException, NoSuchAlgorithmException,
+  private void verify(String token) throws IllegalArgumentException, NoSuchAlgorithmException,
       InvalidKeySpecException, TokenExpiredException, JWTVerificationException, IOException {
     Algorithm algorithm = getAlgorithm(token, getKey());
     JWTVerifier verifier = JWT.require(algorithm).build();
     verifier.verify(token);
+  }
 
-    return true;
+  /**
+   * Checks that access token has sufficient scope to make request
+   * @param token - the access token
+   * @param theRequestDetails - Request details
+   * @return void - throws JWTVerificationException if scope is insufficient
+   */
+  private void checkScopes(String token, RequestDetails theRequestDetails) throws JWTDecodeException, JWTVerificationException {
+    DecodedJWT jwt = JWT.decode(token);
+    Claim claim = jwt.getClaim("scope");
+    String scopes = claim.asString();
+
+    if (scopes == null) throw new JWTVerificationException("Insufficient scope");
+
+    /**
+     * Allow any request if scope includes "system/*.*"
+     * Allow GET request if scope includes "system/*.read"
+     * Allow POST, PUT, DELETE requests if scope includes "system/*.write"
+     */
+    String requestType = theRequestDetails.getRequestType().name();
+    if (scopes.contains("system/*.*") ||
+        (requestType == "GET" && scopes.contains("system/*.read")) ||
+        ((requestType == "POST" || requestType == "PUT" || requestType == "DELETE") &&
+        scopes.contains("system/*.write"))) return;
```

Comment:
In Java String is an object so `==` checks address not data content. I suggest just using the `RequestTypeEnum` here anyway
```suggestion
    RequestTypeEnum requestType = theRequestDetails.getRequestType();
    if (scopes.contains("system/*.*") ||
        (requestType.equals(RequestTypeEnum.GET) && scopes.contains("system/*.read")) ||
        ((requestType.equals(RequestTypeEnum.POST) || requestType.equals(RequestTypeEnum.PUT) || requestType.equals(RequestTypeEnum.DELETE)) &&
        scopes.contains("system/*.write"))) return;
```

Reply:
Should be fixed.